aten/src/ATen/CUDAGeneratorImpl.h

```diff
@@ -84,40 +89,8 @@ namespace at {
  *
  */
 
-
-// Stores state values. Passed as a kernel argument. See "Usage:" above.
-struct PhiloxCudaState {
-  PhiloxCudaState() = default;
-  PhiloxCudaState(const PhiloxCudaState&) = default;
-  // Called if graph capture is not underway
-  PhiloxCudaState(uint64_t seed,
-                  uint64_t offset) {
-    seed_ = seed;
-    offset_.val = offset;
-  }
-  // Called if graph capture is underway
-  PhiloxCudaState(uint64_t seed,
-                  int64_t* offset_extragraph,
-                  uint32_t offset_intragraph) {
-    seed_ = seed;
-    offset_.ptr = offset_extragraph;
-    offset_intragraph_ = offset_intragraph;
-    captured_ = true;
-  }
-
-  // Public members, directly accessible by at::cuda::philox::unpack.
-  // If we made them private with getters/setters, the getters/setters
-  // would have to be __device__, and we can't declare __device__ in ATen.
-  union Payload {
-    uint64_t val;
-    int64_t* ptr;
-  };
-
-  uint64_t seed_;
-  Payload offset_;
-  uint32_t offset_intragraph_;
-  bool captured_ = false;
-};
+// Pulls raw PhiloxCudaState definition into at:: as expected by eager consumers
+#include <ATen/cuda/detail/PhiloxCudaStateRaw.cuh>
```

Comment:
if we use this refactoring strategy (which I think it's much better than copy & paste) - then I we should make the shared headers be as self contained as possible. So move the `namespace at` into PhiloxCudaStateRaw.cuh instead of depending on the surrounding context.

Reply:
Done, but how will that play with your earlier concerns about namespaces?  Tbh im not sure how exactly you wanted me to organize things.

Reply:
With copy-and-paste, you want the minimal contract - even small source code changes can have unexpected consequences so you want compatible but different definitions.

With a shared "header" we can afford the same namespace. (we can still use different namespaces, but it's not a requirement to do so). Since we can afford same namespace, then the problem becomes packaging the shared file in a robust and intuitive way. We should factor the shared definition as a regular header - which means it shouldn't not depend much on the context where it's included. 

Does this make sense?